### PR TITLE
[DEBUGINFRA-724] Migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by sm-powell**
Migrate to GitHub Actions from Jenkins.

Two workflows have been created:

- build-test: Runs when a pull request is created or updated

- post-weekly-release: Runs when
  - Pull request is merged
  - Every Sunday afternoon
  - A version tag is pushed. In this case static analysis will not run, and a draft release is created, e.g. https://github.com/ARM-software/astc-encoder/releases/tag/untagged-5d6fa43431777a71dcb5

### CODE REVIEW

File: .github/workflows/build_test.yaml [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: 'true'
+
+      - name: Build R
+        run: |
+          export CXX=clang++
```
Comment (spcaipers-arm) [inline .github/workflows/build_test.yaml:22]: You can use a job-level env var for that as an alternative.

File: .github/workflows/build_test.yaml [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11'
+
+      - name: Get Python modules
+        run: |
+          python -m pip install --upgrade pip
+          pip install numpy Pillow
```
Comment (spcaipers-arm) [inline .github/workflows/build_test.yaml:52]: You can be more git-friendly by:
```
pip install \
  numpy \
  Pillow
```

File: .github/workflows/build_test.yaml [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
```
Comment (spcaipers-arm) [inline .github/workflows/build_test.yaml:106]: Same as above.

**Review by spcaipers-arm — APPROVED** [review]
In general looks good, there are opportunities to reduce the code by using a matrix job. I see many repeated steps across the jobs.

**Comment by sm-powell** [comment]
Applied standard repo settings: https://confluence.arm.com/display/DTP/GitHub+Repo+Settings

**Review by solidpixel — APPROVED** [review]

**Pull request merged**